go/database/mpt/proof.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package mpt
+
+import (
+	"github.com/Fantom-foundation/Carmen/go/common"
+)
+
+//go:generate mockgen -source proof.go -destination proof_mocks.go -package witness
+
+// rlpEncodedNode is an RLP encoded MPT node.
+type rlpEncodedNode []byte
+
+// proofDb is a database of RLP encoded MPT nodes and their hashes that represent witness proofs.
+type proofDb map[common.Hash]rlpEncodedNode
+
+// WitnessProof represents a witness proof.
+// It contains a database of MPT nodes and their hashes.
+// The proof may be verified by iterating the proof database from the
+// root hash to a terminal entry according to a Path
+// representing either an account address or a storage key.
+// If the proof contains an account or a storage slot for the input address or key,
+// the proof is representing this account or storage.
+type WitnessProof proofDb
```

Comment:
nit: is there are reasonable need for users of proofs to access the proof like a map?
If not, maybe hide this implementation detail like
```suggestion
type WitnessProof struct {
     proofDb
}
```

Reply:
I think it is a good idea. This way, the clients cannot directly modify the proof by adding/removing elements in the map